### DISCUSSION TITLE
Serial Console in v86/buildroot

### DISCUSSION THREAD

**Discussion by giacobe** [copy/v86#1192] [General]
So, I've been messing with v86 and focusing on using Buildroot. I see that the example version on https://copy.sh/v86/?profile=buildroot has a second interface below the captured text interface -- that is all text and is labeled "This is the serial console. Whatever you type or paste here will be sent to COM1"

Can someone help me understand how to activate that? I have successfully instantiated v86, and built my own buildroot.iso (per [humphd's browser-vm](https://github.com/humphd/browser-vm).  I have that working reasonably well. I've messed a bit with some of the buildroot features in the makemenuconfig settings - so I get it the way I want it -- with my code in the VM. I'm wondering if I've simply missed turning on a feature -- because I basically just run it pretty bare-bones.

I was wondering if it was that, or if I had to do something in the inittab maybe -- but I wasn't sure whether just running another tty would be sufficient or if I was missing something else.

Thanks in advance!

**Comment by SuperMaxusa** [comment]
> Can someone help me understand how to activate that?

1. Initialize `xterm.js` in your html:
```html
<!-- run `make build/xterm.js` then -->
<script src="build/xterm.js"></script>
<link rel="stylesheet" href="build/xterm.css">
<!-- or from CDN -->
<script src="https://cdn.jsdelivr.net/npm/xterm@5.2.1/lib/xterm.min.js"></script>
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/xterm@5.2.1/css/xterm.min.css">
```
2. Create an empty `<div>` for the terminal:
```html
<div id="com1"></div>
```
3. Enable serial console in Buildroot: [^1]
4. Add to your `V86` constructor:
```js
serial_container_xtermjs: document.getElementById("com1"),
```

[^1]: for `getty`: https://github.com/humphd/browser-vm/blob/17d220/buildroot-v86/board/v86/rootfs_overlay/etc/inittab#L31-L32 or https://gist.github.com/SuperMaxusa/19a789c9e9b4a69b3237947fdb86d61e#file-rootfs-patch-L41-L45
for `agetty`: https://github.com/copy/v86/blob/298da23/tools/docker/alpine/Dockerfile#L17
